packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1293,14 +1296,18 @@ class Transform extends SingleChildRenderObjectWidget {
   ///    over a given duration.
   Transform.scale({
     Key? key,
-    required double scale,
+    double? scale,
+    double? scaleX,
+    double? scaleY,
     this.origin,
     this.alignment = Alignment.center,
     this.transformHitTests = true,
     this.filterQuality,
     Widget? child,
-  }) : transform = Matrix4.diagonal3Values(scale, scale, 1.0),
-       super(key: key, child: child);
+  })  : assert(!(scale == null && scaleX == null && scaleY == null), "In Transform.scale(), You have to at least provide 'scale' or both of 'scaleX' and 'scaleY'"),
+        assert(scale == null || (scaleX == null && scaleY == null), "In Transform.scale(), you should either provide 'scale' or both of 'scaleX' and 'scaleY' but not all three"),
```

Comment:
Here you can say "If 'scale' is specified, 'scaleX', and 'scaleY' must both be null.".